features/step_definitions/machine.rb

```diff
@@ -0,0 +1,58 @@
+Given(/^I have an IPI deployment$/) do
+  machine_sets = BushSlicer::MachineSet.list(user: admin, project: project("openshift-machine-api"))
+
+  # Usually UPI deployments do not have machineSets
+  if machine_sets.length == 0
+    raise "Not an IPI deployment, abort test."
+  end
+
+  machine_sets.each do | machine_set |
+    unless machine_set.healthy?
+      raise "Not an IPI deployment, abort test."
+    end
+  end
+end
+
+Then(/^the machines should be linked to nodes$/) do
+  machines = BushSlicer::Machine.list(user: admin, project: project("openshift-machine-api"))
+  machines.each do | machine |
+    if machine.node == nil
+      raise "Machine #{machine.name} does not have nodeRef."
+    end
+  end
+end
+
+
+Given(/^I store the number of machines in the#{OPT_SYM} clipboard$/) do | cb_name |
+  machines = BushSlicer::Machine.list(user: admin, project: project("openshift-machine-api"))
+  cb[cb_name] = machines.length
+end
+
+
+Given(/^I store the provisioned machine in the#{OPT_SYM} clipboard$/) do | cb_name |
+  machines = BushSlicer::Machine.list(user: admin, project: project("openshift-machine-api"))
+  step "I run the :get admin command with:", table(%{
+    | resource | machine                                                   |
+    | n        | openshift-machine-api                                     |
+    | sort_by  | {.metadata.creationTimestamp}                             |
+    | template | {{(index .items #{ machines.length - 1 }).metadata.name}} |
+  })
```

Comment:
what is the idea here? You want to get name of first created machine?

Reply:
I need to to get the newest machine, so I run the oc get and sort by creation timestamp to grep the last one. Then machine I get here is the one provisioned when I scale the machineset.

Reply:
Then you can do `cb[cb_name] = machines.max_by(&:created_at)` and avoid the extra `get` step.